dwave/cloud/utils.py

```diff
@@ -801,3 +802,197 @@ def bqm_to_dqm(bqm):
+        qpu_delay_time_per_sample = problem_timing_data['qpu_delay_time_per_sample']
+        ra_with_reinit_delay_time_delta = problem_timing_data['reverse_annealing_with_reinit_delay_time_delta']
+        ra_without_reinit_delay_time_delta = problem_timing_data['reverse_annealing_without_reinit_delay_time_delta']
+        decorrelation_max_nominal_anneal_time = problem_timing_data['decorrelation_max_nominal_anneal_time']
+        decorrelation_time_range = problem_timing_data['decorrelation_time_range']
+        default_readout_thermalization = problem_timing_data['default_readout_thermalization']
+
+    except KeyError as err:
+        if err == 'problem_timing_data':
+            err_msg = f"selected solver is missing required property {err}"
+        elif err == 'version':
+            err_msg = f"selected solver is missing ``problem_timing_data`` field {err}"
+        else:
+            err_msg = f"selected solver is missing ``problem_timing_data`` field " + \
+                      f"{err} required by timing model version {version_timing_model}"
+        raise ValueError(err_msg)
+
+    # Support for sapi timing model versions: 1.0.x
+    version_tuple = tuple(int(i) for i in version_timing_model.split("."))
+    if not version_tuple[0] == 1 and not version_tuple[1] == 0:
+        raise ValueError(f"``estimate_qpu_access_time`` does not currently "
+                         f"support timing model {version_timing_model} "
+                         "used by the selected solver")
+
+    ra_programming_time = 0
+    ra_delay_time = 0
+    if anneal_schedule and anneal_schedule[0][1] == 1:
+        if reinitialize_state:
+            ra_programming_time = ra_with_reinit_prog_time_delta
+            ra_delay_time = ra_with_reinit_delay_time_delta
+        else:
+            ra_programming_time = ra_without_reinit_prog_time_delta
+            ra_delay_time = ra_without_reinit_delay_time_delta
+
+    if programming_thermalization:
+        programming_thermalization_time = programming_thermalization
+    else:
+        programming_thermalization_time = default_programming_thermalization
+
+    programming_time = typical_programming_time + ra_programming_time + \
+                       programming_thermalization_time
+
+    anneal_time = default_annealing_time
+    if annealing_time:
+        anneal_time = annealing_time
+        print(anneal_time)
+    elif anneal_schedule:
+        anneal_time = anneal_schedule[-1][0]
+
+    n = len(readout_time_model_parameters)
+    if n % 2:
+         raise ValueError(f"for the selected solver, ``problem_timing_data`` "
+                          f"property field ``readout_time_model_parameters`` "
+                          f"is not of an even length as required by timing "
+                          f"model version {version_timing_model}")
+
+    q = readout_time_model_parameters[:int(n/2)]
+    t = readout_time_model_parameters[int(n/2):]
```

Comment:
For positive `n`, simpler to use floor division:
```suggestion
    q = readout_time_model_parameters[:n//2]
    t = readout_time_model_parameters[n//2:]
```